addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/GuessParamsThread.java

```diff
@@ -0,0 +1,228 @@
+        this.isAlive = true;
+    }
+
+    private void startGuesser() {
+        // TODO Start a new ParamGuesser
+
+    }
+
+    @Override
+    public void stopScan() {
+        if (paramGuesser != null) {
+            paramGuesser.stop();
+        }
+        stopScan = true;
+        isAlive = false;
+        this.listenner.scanFinshed(site);
+    }
+
+    @Override
+    public void guesserProgress(int percentageComplete, int paramsTried, int totalParams) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void foundParam(String paramName, String reason) {
+        // show param name and reason on output tab .. and for api it should be a CLI output
+
+    }
+
+    @Override
+    public void notifyParamGuessResult(ParamGuessResult result) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void guessComplete(boolean success) {
+        Date finished = new Date();
+        log.info("Param guessing complete: {} on {} at {}", success, site, finished);
+        this.timeTakenInMs = finished.getTime() - started.getTime();
+        stopScan = true;
+        this.isAlive = false;
+        this.listenner.scanFinshed(site);
+    }
+
+    @Override
+    public boolean isRunning() {
+        return isAlive;
+    }
+
+    @Override
+    public void setJustScanInScope(boolean scanInScope) {
+        // Probably wont be used
+
+    }
+
+    @Override
+    public boolean getJustScanInScope() {
+        // Probably wont be used
+        return false;
+    }
+
+    @Override
+    public ListModel<?> getList() {
+        // Not used yet
+        return null;
+    }
+
+    @Override
+    public void reset() {
+        // TODO Add history tab and output tab reset functionality
+
+    }
+
+    @Override
+    public void setScanChildren(boolean scanChildren) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void setScanContext(Context context) {
+        // WOnt be used too I guess
+
+    }
+
+    @Override
+    public void setScanAsUser(User user) {
+        this.scanUser = user;
+    }
+
+    @Override
+    public void setTechSet(TechSet techSet) {
+        // Ignore
+
+    }
+
+    @Override
+    public boolean isStopped() {
+        // TODO Auto-generated method stub
+        return stopScan;
+    }
+
+    @Override
+    public int getProgress() {
+        // TODO Auto-generated method stub
+        return this.progress;
+    }
+
+    @Override
+    public int getMaximum() {
+        return this.guessDone + this.guessTodo;
+    }
+
+    @Override
+    public void pauseScan() {
+        if (paramGuesser != null) {
+            paramGuesser.pause();
+        }
+        this.isPaused = true;
+    }
+
+    @Override
+    public void resumeScan() {
+        if (paramGuesser != null) {
+            paramGuesser.resume();
+        }
+        this.isPaused = false;
+    }
+
+    @Override
+    public boolean isPaused() {
+        return this.isPaused;
```

Comment:
If it’s laudable you might want to use a StopWatch for the time tracking

Reply:
StopWatch?????

Reply:
https://github.com/zaproxy/zaproxy/search?q=stopwatch&type=